fiaas_deploy_daemon/deployer/kubernetes/pod_disruption_budget.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python
+# -*- coding: utf-8
+
+# Copyright 2017-2019 The FIAAS Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import logging
+
+from k8s.client import NotFound
+from k8s.models.common import ObjectMeta, LabelSelector
+from k8s.models.policy_v1_pod_disruption_budget import PodDisruptionBudget, PodDisruptionBudgetSpec
+
+from fiaas_deploy_daemon.retry import retry_on_upsert_conflict
+from fiaas_deploy_daemon.specs.models import AppSpec
+from fiaas_deploy_daemon.tools import merge_dicts
+
+LOG = logging.getLogger(__name__)
+
+
+class PodDisruptionBudgetDeployer(object):
+    def __init__(self, owner_references, extension_hook):
+        self._owner_references = owner_references
+        self._extension_hook = extension_hook
+
+    @retry_on_upsert_conflict
+    def deploy(self, app_spec: AppSpec, selector: dict[str, any], labels: dict[str, any]):
+        if app_spec.autoscaler.min_replicas == 1 or app_spec.autoscaler.max_replicas == 1:
```

Comment:
I just want to double check this should be `or` ? The discussions I heard revolved around singletons, which would be `and`. But this might be correct regardless?

Reply:
We don't want a PDB if there is any circumstance where the app might have a single replica, as that would stall eviction of the pod

Reply:
Yes, I think for any case where there can be only one replica (replicas.min=replicas.max=1, with singleton enabled or not; replicas.min=1; replicas.max=1) there shouldn't be a PDB with maxUnavailable=1